vpc.tf

```
@@ -6,67 +6,75 @@ AZ), and default routes for them in the private subnets.
 */
 
 resource "aws_subnet" "public" {
-  count = var.create_networking_config ? length(var.public_subnet_cidrs): 0
-  cidr_block = var.public_subnet_cidrs[count.index]
-  vpc_id = var.vpc_id
+  count                   = var.create_networking_config ? length(var.public_subnet_cidrs) : 0
+  cidr_block              = var.public_subnet_cidrs[count.index]
+  vpc_id                  = var.vpc_id
   map_public_ip_on_launch = true
-  availability_zone = count.index % 2 == 0 ? "${var.region}a" : "${var.region}b"
-  tags = merge({
+  availability_zone       = count.index % 2 == 0 ? "${var.region}a" : "${var.region}b"
+  tags                    = merge({
     Name = "mwaa-${var.environment_name}-public-subnet-${count.index}"
   }, var.tags)
 }
 
 resource "aws_subnet" "private" {
-  count = var.create_networking_config ? length(var.private_subnet_cidrs): 0
-  cidr_block = var.private_subnet_cidrs[count.index]
-  vpc_id = var.vpc_id
+  count                   = var.create_networking_config ? length(var.private_subnet_cidrs) : 0
+  cidr_block              = var.private_subnet_cidrs[count.index]
+  vpc_id                  = var.vpc_id
   map_public_ip_on_launch = false
-  availability_zone = count.index % 2 == 0 ? "${var.region}a" : "${var.region}b"
-  tags = merge({
+  availability_zone       = count.index % 2 == 0 ? "${var.region}a" : "${var.region}b"
+  tags                    = merge({
     Name = "mwaa-${var.environment_name}-private-subnet-${count.index}"
   }, var.tags)
 }
 
 resource "aws_eip" "this" {
-  count = var.create_networking_config ? length(var.public_subnet_cidrs): 0
-  vpc = true
-  tags = merge({
+  count = var.create_networking_config ? length(var.public_subnet_cidrs) : 0
+  vpc   = true
+  tags  = merge({
     Name = "mwaa-${var.environment_name}-eip-${count.index}"
   }, var.tags)
 }
 
 resource "aws_nat_gateway" "this" {
-  count = var.create_networking_config ? length(var.public_subnet_cidrs): 0
+  count         = var.create_networking_config ? length(var.public_subnet_cidrs) : 0
   allocation_id = aws_eip.this[count.index].id
-  subnet_id = aws_subnet.public[count.index].id
-  tags = merge({
+  subnet_id     = aws_subnet.public[count.index].id
+  tags          = merge({
     Name = "mwaa-${var.environment_name}-nat-gateway-${count.index}"
   }, var.tags)
 }
 
+resource "aws_internet_gateway" "this" {
+  count  = var.create_networking_config && var.internet_gateway_id==null ? 1 : 0
+  vpc_id = var.vpc_id
+  tags   = merge({
+    Name = "mwaa-${var.environment_name}-internet-gateway"
+  }, var.tags)
+}
+
 resource "aws_route_table" "public" {
-  count = var.create_networking_config ? 1: 0
+  count  = var.create_networking_config ? 1 : 0
   vpc_id = var.vpc_id
   route {
     cidr_block = "0.0.0.0/0"
-    gateway_id = var.internet_gateway_id
+    gateway_id = var.internet_gateway_id==null ? var.internet_gateway_id : aws_internet_gateway.this[0].id
```

hier war das != schon richtig oder du drehst das um
```suggestion
    gateway_id = var.internet_gateway_id==null ? aws_internet_gateway.this[0].id : var.internet_gateway_id
```